cron.py

```diff
@@ -872,6 +895,48 @@ async def check_hibp():
     LOG.d("Done checking HIBP API for aliases in breaches")
 
 
+def notify_hibp():
+    """
+    Send aggregated email reports for HIBP breaches
+    """
+    for user in User.query.all():
+        new_breaches = (
+            Alias.query.filter(Alias.user_id == user.id)
+            .filter(Alias.hibp_breaches_not_notified_user)
+            .all()
+        )
+
+        if len(new_breaches) == 0:
+            return
+
+        LOG.d(f"Send new breaches found email to user {user}")
+
+        send_email(
+            user.email,
+            f"You were in a data breach",
+            render(
+                "transactional/hibp-new-breaches.txt",
+                user=user,
+                breached_aliases=new_breaches,
+            ),
+            render(
+                "transactional/hibp-new-breaches.html",
+                user=user,
+                breached_aliases=new_breaches,
+            ),
+        )
+
+        for alias in new_breaches:
+            alias.hibp_breaches_notified_user = (
+                alias.hibp_breaches_notified_user
+                + alias.hibp_breaches_not_notified_user
+            )
+            alias.hibp_breaches_not_notified_user = []
+            db.session.add(alias)
```

Comment:
is `db.session.add(alias)` needed?

Reply:
Hmm, I hadn't tested it without but it made sense to me to explicitly add it given we just changed it and we want it to be part of the commit.

Reply:
Oh I see, I always thought of "db.session.add()" as adding an object to the database but not the active session. I think your understanding is more correct 👍